flink/src/main/scala/ru/itclover/tsp/PatternsSearchJob.scala

```diff
@@ -182,7 +182,12 @@ object PatternsSearchJob {
             .fromEither(pGenerator.build(p.sourceCode, toleranceFraction, fieldsTags))
             .leftMap(err => List(s"PatternID#${p.id}, error: ${err.getMessage}"))
             .map(
-              p => (new IdxMapPattern(p._1)(segmentize).asInstanceOf[Pattern[E, AnyState[Segment], Segment]], p._2)
+              p =>
+                p match {
+                  case (null, _) => sys.error("Pattern is nulle")
+                  case _ =>
+                    (new IdxMapPattern(p._1)(segmentize).asInstanceOf[Pattern[E, AnyState[Segment], Segment]], p._2)
+                }
```

Comment:
Oh, I forgot to remove this part (I've added this for testing since pattern generator returned null for some cryptic reason - possibly due to implicit decoder import)